basis/io/directories/directories.factor

```diff
@@ -93,6 +85,18 @@ M: object copy-file
 : copy-files-into ( files to -- )
     '[ _ copy-file-into ] each ;
 
+! Moving and renaming files
+HOOK: move-file io-backend ( from to -- )
+
+M: object move-file ( from to -- )
+    [ normalize-path ] bi@ dupd copy-file delete-file ;
```

Comment:
Because it delegates to copy-file and delete-file, presumably if there is file path sanitization required, those will have to do it also so keep this simple.

> On Jun 28, 2016, at 9:19 AM, Alexander Iljin <notifications@github.com> wrote:
> 
> In basis/io/directories/directories.factor:
> 
> > @@ -93,6 +85,18 @@ M: object copy-file
> >  : copy-files-into ( files to -- )
> >      '[ _ copy-file-into ] each ;
> >  
> > +! Moving and renaming files
> > +HOOK: move-file io-backend ( from to -- )
> > +
> > +M: object move-file ( from to -- )
> > +    [ normalize-path ] bi@ dupd copy-file delete-file ;
> Why not? This method is callable by user, to presumably it should sanitize/preprocess the parameters? Or is there a wrapper that does this?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>